langchain-interpreter/src/langchain_interpreter/parser.py

```
@@ -0,0 +1,250 @@
+    yaml_lines = lines[1:end_index]
+    yaml_content = "\n".join(yaml_lines)
+
+    if not yaml_content.strip():
+        # Empty frontmatter
+        return AgentMetadata(), end_index + 1
+
+    # Parse YAML
+    try:
+        yaml_data = yaml.safe_load(yaml_content)
+    except yaml.YAMLError as e:
+        raise AFMParseError(f"Invalid YAML in frontmatter: {e}")
```

Ruff (0.14.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)